src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc

```diff
@@ -808,6 +818,11 @@ const std::shared_ptr<BundleLocations> &LeaseStatusTracker::GetPreparedBundleLoc
   return preparing_bundle_locations_;
 }
 
+const std::unordered_map<NodeID, std::vector<std::shared_ptr<const BundleSpecification>>>
+    &LeaseStatusTracker::GetGroupedPreparedBundleLocations() const {
+  return grouped_preparing_bundle_locations_;
```

Comment:
Why don't we just construct on the fly? In this way, we can avoid having a new separate field?

Reply:
Updated.